pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -209,6 +209,9 @@ type XdsConnection struct {
 
 	// Time of last push failure.
 	LastPushFailure time.Time
+
+	// mutex protects changes to this XDS connection
+	mutex sync.RWMutex
```

Comment:
Please put the mutex before the fields it guards; if the mutex guards all fields in the struct then the mutex should be the first field in the struct. And nit, please name it `m` or `mu` or similar, rather than `mutex` (which IMO is overly long given its context/usage).